src/wasm-emscripten.cpp

```diff
@@ -308,6 +324,31 @@ void AsmConstWalker::addImport(Name importName, std::string baseSig) {
   wasm.addImport(import);
 }
 
+AsmConstWalker fixEmAsmConstsAndReturnWalker(Module& wasm) {
```

Comment:
It looks like this walker runs twice, once in wasm-emscripten-finalize tool and again in wasm-link-metadata tool. Is this redundancy intended, or can't be helped? Asking because I have similar concerns for other stuff; I need to generate some emscripten glue and also some metadata based on the same walker. In the current setup, the two tools are separate, so should the walker always run twice?